Source/loadsave.cpp

```diff
@@ -824,6 +916,36 @@ void ConvertLevels()
 	tbuff = _tbuff;
 }
 
+void LoadHotkeys()
+{
+	LoadHelper file("hotkeys");
+	if (!file.isValid())
+		return;
+
+	for (size_t i = 0; i < sizeof(plr[myplr]._pSplHotKey) / sizeof(plr[myplr]._pSplHotKey[0]); i++) {
+		plr[myplr]._pSplHotKey[i] = file.nextLE32();
+	}
+	for (size_t i = 0; i < sizeof(plr[myplr]._pSplTHotKey) / sizeof(plr[myplr]._pSplTHotKey[0]); i++) {
+		plr[myplr]._pSplTHotKey[i] = file.nextByte();
+	}
+	plr[myplr]._pRSpell = file.nextLE32();
+	plr[myplr]._pRSplType = file.nextByte();
+}
+
+void SaveHotkeys()
+{
+	SaveHelper file("hotkeys", sizeof(plr[myplr]._pSplHotKey) + sizeof(plr[myplr]._pSplTHotKey) + 5);
+

```

Comment:
why that ugly 5 instead of sizeofs? :P

Reply:
It's hardcoded to save as a 32bit and an 8bit value so sizeof() could actually give us the wrong value. Actually, that made me realize that the same should be done for the two others :o

Reply:
Better? https://github.com/diasurgical/devilutionX/pull/1124

Reply:
Hmm, just curious. Why did you save them to a config file instead of just saving the data at the end of the save file?

You could have done:
`ISave(plr[myplr]._pSplTHotKey[i]`

Reply:
There are explicit checks on the size of the `hero` file, so changing the size would cause it to show as an invalid save in vanila.